File: java_solutions/1470.shuffle-the-array.java
/*
 * @lc app=leetcode id=1470 lang=java
 *
 * [1470] Shuffle the Array
 */

class Solution {
    public int[] shuffle(int[] nums, int n) {
        return IntStream.range(0, 2 * n)
            .map(i -> i % 2 == 0 ? nums[i / 2] : nums[n + i / 2])
            .toArray();
    }
}

File: kotlin_solutions_1/1470.shuffle-the-array.kt
/*
 * @lc app=leetcode id=1470 lang=kotlin
 *
 * [1470] Shuffle the Array
 */

class Solution {
    fun shuffle(nums: IntArray, n: Int): IntArray = IntArray(2 * n) { if (it % 2 == 0) nums[it / 2] else nums[n + it / 2] }
}